src/encoded/types/cart.py

```diff
@@ -96,3 +101,50 @@ def get_or_create_cart_by_user(context, request):
         '@type': ['result'],
         '@graph': carts or [cart]
     }
+
+
+@view_config(context=Cart.Collection, request_method='PUT', permission='save-carts', name='put-cart')
+def create_cart_by_user(context, request):
+    userid = get_userid(request)
+    user = _get_user(request, userid)
+    blocked_statuses = ['deleted'] if 'group.admin' not in request.effective_principals else []
+    carts = get_cart_objects_by_user(request, userid, blocked_statuses=blocked_statuses)
+    cart_atids = [cart['@id'] for cart in carts]
+    cart_status = request.json.get('status')
+    cart_name = request.json.get('name', '').strip()
+    # User writing a new cart; check for cart overflow and naming conflicts
+    countable_carts = []
+    if cart_status != 'disabled':
+        countable_carts = [
+            cart
+            for cart in carts
+            if cart['status'] not in ['disabled', 'deleted']
+        ]
+    if len(countable_carts) >= CART_USER_MAX:
+        msg = 'Users cannot have more than {} carts'.format(CART_USER_MAX)
+        raise HTTPBadRequest(explanation=msg)
+    name_conflict = any(
```

Comment:
Sorry one more change with this.

I realized that we are just filtering a list here and if anything is remaining in the list when you check `if name_conflict` it will evaluate to True.

So either:
1. We remove `any` and just evaluate the list comprehension.
2. We remove the list comprehension and pass `any` a generator comprehension.

Potentially the benefit of #2 is that the expression will stop evaluating as soon as it reaches a true value, instead of iterating through the whole list.

I don't really have a strong preference either way but we shouldn't leave it how it is now.